src/include/platform/Mdns.h

```diff
@@ -0,0 +1,167 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include "core/CHIPError.h"
+#include "inet/IPAddress.h"
+#include "inet/InetInterface.h"
+
+namespace chip {
+namespace DeviceLayer {
+
+static constexpr uint8_t kMdnsNameMaxSize  = 32;
+static constexpr uint8_t kMdnsTypeMaxSize  = 32;
+static constexpr uint16_t kMdnsTextMaxSize = 64;
+
+enum class MdnsServiceProtocol : uint8_t
+{
+    kMdnsProtocolUdp = 0,
+    kMdnsProtocolTcp,
+    kMdnsProtocolUnknown = 255,
+};
+
+struct TextEntry
+{
+    const char * mKey;
+    const uint8_t * mData;
```

Comment:
Are key and data really different types?

Reply:
Yes, RFC requires the key to be a readable ascii string but data can contain arbitrary binary data.